src/main/java/trackitnus/model/ModelManager.java

```diff
@@ -276,6 +276,11 @@ public void addLesson(Lesson module) {
         trackIter.addLesson(module);
     }
 
+    @Override
+    public boolean canAddModule() {
+        return getFilteredModuleList().size() < NUM_MODULE_LIMIT;
+    }
+
```

Comment:
I feel that this function should also check if the module alr exists in trackit? Or maybe rename the function to `hasEnoughSpace` to make it clearer